packages/flutter/lib/src/rendering/object.dart

```diff
@@ -82,6 +82,22 @@ class PaintingContext extends ClipContext {
   /// The [estimatedBounds] rectangle is in the [canvas] coordinate system.
   final Rect estimatedBounds;
 
+  /// Paint [child] with this painting context, then convert to a raster and detach all
+  /// children from this layer.
```

Comment:
This documentation could be a little more extensive. One thing to mention would be that the caller is responsible for actually drawing the returned image. Otherwise my understanding is that it will no show up, which may not what people expect from a method that's called "paint".